docs/explanation/dss-arch.rst

```diff
@@ -0,0 +1,153 @@
+
+MLflow
+^^^^^^
+
+MLflow operates in `local mode <https://mlflow.org/docs/latest/tracking.html#other-configuration-with-mlflow-tracking-server>`_, meaning that metadata and artefacts for each run are, by default, recorded to a local directory named `mlruns`. 
+This directory is backed by a persistent volume, mounted to a Hostpath directory of the MLflow Kubernetes deployment.
+
+.. _container_layer:
+
+Containerisation
+~~~~~~~~~~~~~~~~
+
+DSS needs a container orchestration solution. 
+DSS relies on MicroK8s, a lightweight Kubernetes distribution.
+
+Therefore, MicroK8s needs to be deployed before installing DSS on the host machine. 
+It must be configured with the storage add-on. 
+This is required to use Hostpath storage in the cluster. 
+
+DSS can run with or without the use of GPUs.
+If :ref:`gpu_support` is needed, MicroK8s can be configured with the desired `GPU add-on <https://microk8s.io/docs/addon-gpu>`_. 
+
+.. _gpu_support:
+
+GPU support
+^^^^^^^^^^^
+
+DSS does not automatically install the tools and libraries required for running GPU workloads.
+To do so, it relies on MicroK8s, or any other Kubernetes deployment used as the orchestration solution.
+
+DSS is designed to support the deployment of containerised GPU workloads on NVIDIA GPUs. 
+MicroK8s simplifies the GPU access and usage through the `NVIDIA GPU Operator <NVIDIA Operator_>`_. 
+
+.. caution::
+   GPUs from other silicon vendors rather than NVIDIA can be configured. However, its functionality is not guaranteed.
+ 
+Storage
+^^^^^^^
+
+DSS expects a default `storage class <https://kubernetes.io/docs/concepts/storage/storage-classes/>`_ in the Kubernetes deployment, which is used to persist Jupyter Notebooks and MLflow artefacts.   
+In MicroK8s, the Hostpath storage add-on is chosen, used to provision Kubernetes' *PersistentVolumeClaims* (`PVCs <https://kubernetes.io/docs/concepts/storage/persistent-volumes/>`_). 
+
+A shared PVC is used across all Jupyter Notebooks to share and persist data. 
+MLflow also uses its dedicated PVC to store the logged artefacts.
+This is the DSS default storage configuration and cannot be altered.
+
+This choice ensures that all storage is backed up on the host machine in the event of MicroK8s restarts.
+
+.. note::
+   By default, users can access this storage anytime under `/var/snap/microk8s/common/default-storage`. 
+
+The following diagram summarises the DSS storage:
+
+.. figure:: https://assets.ubuntu.com/v1/2130fd66-dss_storage.png
+   :width: 800px
+   :alt: text
+
+   Storage overview
+
+.. _os_layer:
+
+Operating system
+~~~~~~~~~~~~~~~~
+
+DSS is native on Ubuntu, being developed, tested and validated on it. 
+
+Moreover, the solution can be used on any Linux distribution, Windows through Windows Subsystem Linux (WSL) or MacOS through Multipass, 
+but the GPU support is at risk, depending on the GPU operator. 
+
+Namespace configuration
+-----------------------
+
+DSS has a dedicated namespace running on MicroK8s. 
```

Comment:
This sounds like we have a concept of `dss namespaces` which is not true because those are just kubernetes namespaces.